gazelle/python/generate.go

```diff
@@ -463,6 +470,20 @@ func hasEntrypointFile(dir string) bool {
 	return false
 }
 
+// hasNameEqualsMain determines if the file contains 'if __name__ == "__main__"'.
+func hasNameEqualsMain(path string) bool {
+	searchString := `if __name__ == ['"]__main__['"]:`
+	bytesContents, err := os.ReadFile(path)
```

Comment:
Right now we are reading the python files once to get the imports and here it would be for once more to check if there is an `if __name__ == "main"` which may not scale well in super large repos because we have now twice the number of files to process.

Reply:
Let me see how the read to get the import works. Maybe I can figure out a way to not read it twice.